workflows/cluster_plots/__main__.py

```diff
@@ -0,0 +1,47 @@
+import os
+
+import luigi
+from plotting import ClusterPlottingTask
+
+from birdclef.knn_labels import (
+    compute_embedding_2d,
+    get_knn_labels,
+    get_label_agreement,
+    get_subset_pdf,
+    plot_distances,
+    plot_embedding,
+    write_plots_to_disk,
+)
+from birdclef.utils import get_spark
+
+
+class ClusterPlotTask(luigi.WrapperTask):
+    path_prefix = luigi.Parameter()
+
+    def requires(self):
+        spark = get_spark(memory="16g")
+        df = spark.read.parquet(
+            "../data/processed/birdclef-2022/birdnet-embeddings-with-neighbors/v1"
+        )
+        labeled_neighborhood = get_knn_labels(df).cache()
+
+        for i in range(len(df)):
+            cluster_plots = ClusterPlottingTask(
+                path_prefix=self.path_prefix,
+                df=df,
+                labeled_neighborhood=labeled_neighborhood,
+                index=i,
+            )
+            yield cluster_plots
```

Comment:
Take a look at the Pull and Push modules that @ChrisHayduk wrote here.  We probably want to pull the processed dataset down from the bucket and make it a dependency of the ClusterPlottingTask. You should be able to pass this cluster_plots task as a dynamic requires into a Push task, which will copy the folder up into the bucket at the specified location and add a _SUCCESS file so that luigi knows that it doesn't need to be rerun.

https://github.com/dsgt-birdclef/birdclef-2023/blob/ef4313b36e4cde87789b6d3ee367188c0774bcd1/workflows/convert_audio/__main__.py#L39-L45

Ah, also I noticed that you put the spark job in the metatask, which will make it hard to parallelize the task here. Does it actually run more than one task at a time in the UI? I think we can basically scale up the number of workers to half the number of cpus otherwise.

We should make the input of the `ClusterPlottingTask` the location to the parquet file, and then have luigi spin up a new process to run that task. We can't serialize the dataframe here because it's handled by another process. Just push the df/labeled_neighborhood stuff down into the task to resolve this.

Here's the original Spark paper on resilient distributed datasets (RDD) in case you're curious: https://www.usenix.org/system/files/conference/nsdi12/nsdi12-final138.pdf It's a powerful tool.